### DISCUSSION TITLE
Can't get Lit or components to work - help!

### DISCUSSION THREAD

**Discussion by andrebaerlocher** [lit/lit#4176] [Q&A] [answered]
I am trying to get into Lit and struggling with the basics, apparently. 

My html (edit: which isn't displayed, if it's code - goshdammit, can't even get that to work properly, even not as code):
~~~
<!DOCTYPE html>
<head>
    <script type="module" src="./my-el.js"></script>
</head>
<body>
  <my-element></my-element>
</body>
</html>
~~~

My lit.js:
~~~
import {LitElement, html} from 'lit';

export class MyElement extends LitElement {
  render () {
    return html`
        <p>Henlo</p>
    `;
  }
        
}

customElements.define('my-element', MyElement);
~~~
My result:
![image](https://github.com/lit/lit/assets/36181845/ebc467c8-290a-45c8-b10a-dadfaab8d716)

What am I doing wrong?
I have installed Lit via npm, checked the version to see whether I have really REALLY installed it
![image](https://github.com/lit/lit/assets/36181845/73aa4336-adbe-4a81-8190-858caa1694b6)

**Comment by bennypowers** [comment]
Your result is correct. You should see the paragraph rendered in shadow DOM.

More about shadow DOM: https://lit.dev/docs/components/shadow-dom/

**Reply by bennypowers** [reply]
Try using a development server like web dev server

From your project directory, run this in terminal:

```
npx @web/dev-server --open --watch
```

More about web dev server: https://modern-web.dev/docs/dev-server/overview/

**Reply by andrebaerlocher** [reply]
I did, yet still no shadow root
![image](https://github.com/lit/lit/assets/36181845/d03227c1-6df4-4dcc-86c7-eeaaea7e8b9d)

**Reply by bennypowers** [reply]
I can see that you received an error in the dev tools console; could you paste the text of the error message here?

**Reply by andrebaerlocher** [reply]
I did, my gosh, didn't notice! 
`Uncaught TypeError: Failed to resolve module specifier "lit". Relative references must start with either "/", "./", or "../".`
/Edit: I am very thankful for your help, appreciate it a lot!

**Reply by bennypowers** [reply, marked as answer]
No problem

There are two main options to solve this error

The first is called import maps. You can generate an import map for your page at http://generator.jspm.dev and it will work in production.

Another option is called node resolution. You can turn it on by adding `--node-resolution` to the wds command. This will only work for the dev server: for production, you'll need to use a bundler like esbuild, rollup, or webpack. Configuration of those bundlers is complex, you should seek support for them in their dedicated channels.

**Reply by andrebaerlocher** [reply]
Feeling a bit dumb: Can't open the link you posted. /Edit: I'm dumb, link isn't supposed to be opened, is it?

And other than that:
![image](https://github.com/lit/lit/assets/36181845/95707743-8a03-4dd9-935a-501170c950b0)

**Reply by bennypowers** [reply]
https://generator.jspm.io/#U2VhYGAIzSvJLMlJTWEAACKwkqAOAA

Enter `lit` under dependencies and copy the HTML boilerplate to your page

Add your script tag and your custom element

**Reply by andrebaerlocher** [reply]
It works, you're amazing! Many, many thanks!
/Edit: Not sure if you'll ever see this, but Lit is amazing and it works thanks to you! Ever. So. Happy!
